src/main/java/com/aetherteam/aetherii/entity/passive/Aerwhale.java

```diff
@@ -0,0 +1,639 @@
+        this.goalSelector.addGoal(5, new Aerwhale.RandomFloatAroundGoal(this));
+    }
+
+    public static AttributeSupplier.Builder createMobAttributes() {
+        return Mob.createMobAttributes()
+                .add(Attributes.MAX_HEALTH, 20.0)
+                .add(Attributes.FLYING_SPEED, 0.2)
+                .add(Attributes.MOVEMENT_SPEED, 0.2)
+                .add(Attributes.CAMERA_DISTANCE, 8.0)
+                .add(Attributes.STEP_HEIGHT, 0.4);
+    }
+
+    @Override
+    protected void checkFallDamage(double y, boolean onGround, BlockState state, BlockPos pos) {}
+
+    @Override
+    protected void defineSynchedData(SynchedEntityData.Builder builder) {
+        super.defineSynchedData(builder);
+    }
+
+    // @Nullable
+    // @Override
+    // public AgeableMob getBreedOffspring(ServerLevel level, AgeableMob otherParent) {
+    //     return AetherIIEntityTypes.AERWHALE.get().create(level, EntitySpawnReason.BREEDING);
+    // }
+
+    /**
+     * Aerwhales can spawn if {@link Mob#checkMobSpawnRules(EntityType, LevelAccessor, MobSpawnType, BlockPos, RandomSource)} is true, if they aren't spawning in fluid,
+     * if they are spawning at a light level above 8, if they are spawning in view of the sky, and they spawn with a random chance of 1/40.
+     *
+     * @param aerwhale The {@link Aerwhale} {@link EntityType}.
+     * @param level    The {@link LevelAccessor}.
+     * @param reason   The {@link EntitySpawnReason} reason.
+     * @param pos      The spawn {@link BlockPos}.
+     * @param random   The {@link RandomSource}.
+     * @return Whether this entity can spawn, as a {@link Boolean}.
+     */
+    public static boolean checkAerwhaleSpawnRules(EntityType<? extends Aerwhale> aerwhale, LevelAccessor level, EntitySpawnReason reason, BlockPos pos, RandomSource random) {
+        return level.getFluidState(pos).is(Fluids.EMPTY)
+                && level.getRawBrightness(pos, 0) > 8
+                && level.canSeeSky(pos)
+                && (reason != EntitySpawnReason.NATURAL || random.nextInt(40) == 0);
+    }
+
+    @Override
+    public boolean onClimbable() {
+        return false;
+    }
+
+    /**
+     * The purpose of this method override is to fix the weird movement from flying mobs.
+     *
+     * @param vector The {@link Vec3} for travel movement.
+     */
+    @Override
+    public void travel(Vec3 vector) {
+        float speed = (float)this.getAttributeValue(Attributes.FLYING_SPEED) * 5.0f / 3.0f;
+        this.travelFlying(vector, speed, speed, speed);
+    }
+
+    @Override
+    public void tick() {
+        super.tick();
+    }
+
+    @Override
+    public LivingEntity getControllingPassenger() {
+        return this.getFirstPassenger() instanceof Player player? player : super.getControllingPassenger();
+    }
+
+    /**
+     * [CODE COPY] - {@link HappyGhast#getRiddenInput}
+     */
+    @Override
+    protected Vec3 getRiddenInput(Player player, Vec3 travelVector) {
```

Comment:
ride-related code shouldn't be necessary; aerwhale won't be allowed to be ridden